spring-cloud-stream-binders/spring-cloud-stream-binder-test/src/main/java/org/springframework/cloud/stream/binder/PartitionCapableBinderTests.java

```diff
@@ -333,4 +329,9 @@ protected String getPubSubEndpointRouting(AbstractEndpoint endpoint) {
 
 	protected abstract String getClassUnderTestName();
 
+	protected AbstractEndpoint extractEndpoint(Binding<MessageChannel> binding) {
+		Assert.isInstanceOf(DefaultBinding.class, binding, "Binding not of expected type, found " + binding.getClass().getName() + " instead");
```

Comment:
We can remove it for leniency. Maybe add a check for the endpoint field instead. Just to give a more meaningful error.